src/util/list.c

```diff
@@ -56,14 +57,14 @@ list_t *create_list(size_t elementSize, freeFunction freeFn)
     list->logicalLength = 0;
     list->elementSize = elementSize;
     list->head = list->tail = NULL;
+    list->current = NULL;
```

Comment:
Interesting addition. Thinking about this little corner of EPANET, I no longer consider lists the right container for this data. An array would work just as well and you can use your index on them. However, given that the number of demand categories is likely to be so small it's hardly worth messing around with.